include/Spectra/LinAlg/Orthogonalization.h

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2020 Netherlands eScience Center <f.zapata@esciencecenter.nl>
+//
+// This Source Code Form is subject to the terms of the Mozilla
+// Public License v. 2.0. If a copy of the MPL was not distributed
+// with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef SPECTRA_ORTHOGONALIZATION_H
+#define SPECTRA_ORTHOGONALIZATION_H
+
+#include <Eigen/Core>
+
+namespace Spectra {
+
+/// Gram-Schmidt process to orthoganlize a given basis A. Each column correspond
+/// to a vector on the basis.
+/// The start index indicates from what vector to start
+/// the orthogonalization, 0 is the default.
+/// Warnings: Starting the normalization at index n imples that the previous
+/// n-1 vectors are already orthonormal.
+
+template <typename Scalar>
+class Orthogonalization
```

Comment:
can we call it GramSchmidt or ModifiedGrahmSchmidt ? because we may still do a QR or any other methods to orthgonalize a matrix

Reply:
maybe a function would be better than a class also. 
Or overloading the call operator so that we can write
```
gs = GrahmSchindt();
Aortho = gs(A, 5);
```

I don't know :) I just find the `Orthogonalization.orthonormalisation()` strange to write :)

Reply:
@NicoRenaud  I have moved the functions to `Orthonormalize` class and there I have three methods:
* `modfied_gramscmidt`
* `twice_modified_gramschmidt`
* `QR`

Is that a good interface for you? :)

Reply:
there is no twice modified gramschmidt

Reply:
Yes there is, there are 3 public method to perform orthogonalization. Check: https://github.com/NLESC-JCER/spectra/blob/gramschmidt/include/Spectra/LinAlg/Orthogonalization.h